gen3/bin/iam-serviceaccount.sh

```diff
@@ -0,0 +1,604 @@
+#!/bin/bash
+# 
+# After EKS 1.13, AWS allows OIDC identity provider to interact with serviceaccount in k8s
+# this would be helpful when manipulating permissions for pod directly.
+# With this implementation, there should not be the need to creeate users and 
+# key secret pair to access things in AWS, but insted roles will be applied to 
+# pods directly through service accounts.
+# Additionally, we could potentially remove certain roles attached to k8s workers.
+#
+# This script is intended to automate certain steps for this implementation, and ease 
+# the burden of creating policy and roles directly on the console or through awscli
+#
+# For more information visit the folowing link
+# https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/
+
+source "${GEN3_HOME}/gen3/lib/utils.sh"
+gen3_load "gen3/lib/kube-setup-init"
+
+SCRIPT=$(basename ${BASH_SOURCE[0]})
+ACOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+
+LOGS_SCRIPT_FILE="/tmp/iam-serviceaccount.log"
+
+echo "logging at ${LOGS_SCRIPT_FILE}"
+echo > ${LOGS_SCRIPT_FILE}
+
+##
+#
+# function to just print and error and exit 
+#
+# @arg string message to print
+#
+##
+function print_error_and_exit() {
+  local message="${1}"
+  exit 2
```

Comment:
no print?

Reply:
Going to take this function out, I thought of replacing `echo`s around with the function, but seemed like more work, and not very practical, will maybe find a better way to do it.